### PR TITLE
Improvements and fixes found during microos poc

### PR DESCRIPTION
**Pull request by jreidinger**
## Problem

During creating POC for microos variant of d-installer few possible improvements found like better logging or ensuring that some clients are run with proper target.

## Solution

separate these improvements and add it now to master.

## Testing

- Tested as part of microos poc.
- manually tested also with master installation.

### CODE REVIEW

File: service/lib/dinstaller/software.rb [hunk 1]
```diff
@@ -129,6 +127,16 @@ def finish(_progress)
 
   private
 
+    def solve_dependencies
+      res = Yast::Pkg.PkgSolve(unused = true)
+      logger.info "solver run #{res.inspect}"
+      if !res
```
[the remark this answers is not in the document]
Comment (jreidinger) [reply service/lib/dinstaller/software.rb]: rubocop agrees with you

**Comment by jreidinger** [comment]
@imobachgs testing works. So it is ready for final review and possibly merge.

File: service/lib/dinstaller/software.rb [hunk 2]
```diff
@@ -82,24 +82,22 @@ def probe(progress)
     end
 
     def propose
+      Yast::Pkg.TargetFinish # ensure that previous target is close
```
Comment (imobachgs) [inline service/lib/dinstaller/software.rb:85]: np: closed

**Review by imobachgs — APPROVED** [review]
It looks good. Thanks!

**Pull request merged**